### PR TITLE
Palette

### PR DESCRIPTION
**Pull request by davidpmccormick**
This updates the palette to display the correct values (as delivered by C&T), as well as splitting the colours out into 'primary' and 'secondary' sections.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
Any chance of a screenshot to give a little more context?

File: client/scss/styleguide/_styleguide.scss [hunk 1]
```diff
@@ -25,30 +22,72 @@
...
-  background: color('white');
-  padding: 5px;
-  max-width: 190px;
-  border-radius: 3px;
-  border: 1px solid color('smoke');
+.styleguide__palette__hyphen {
+  visibility: hidden;
+}
+
+.styleguide__palette__name {
```
Comment (jamesgorrie) [inline client/scss/styleguide/_styleguide.scss]: Not sure why we need this namespace... BEM isn't really meant for name-spacing.
To namespace we could use the more traditional SASS:
```SASS
.styleguide {
  .palette__hyphen {}
}
```

which I suppose denotes that we only use it in styleguide, but I'm not sure where the value in that information lies?

**Comment by davidpmccormick** [comment]
screen shot 2016-10-28 at 15 40 23

File: client/js/components/palette.js [hunk 2]
```diff
@@ -1,20 +1,47 @@
...
-    `;
-
-    el.appendChild(descriptionEl.querySelector('.styleguide__hex'));
+  const palette = computedStyle.map((str) => {
+    const split1 = str.indexOf('|');
+    const split2 = str.indexOf(',');
+    const cssName = str.slice(0, split1);
+    const name = cssName.replace(/-/g, '<span class="styleguide__palette__hyphen">-</span>');
+    const hex = str.slice(split1 + 1, split2);
+    const attribution = str.slice(split2 + 1);
```
Comment (jamesgorrie) [inline client/js/components/palette.js]: Let's maybe have a chat about how to potentially get it directly into the template, it's an interesting challenge.

File: client/js/components/palette.js [hunk 3]
```diff
@@ -1,20 +1,47 @@
...
+    const wrappingEl = document.createElement('div');
+    const headingEl = document.createElement('h3');
+    headingEl.className = 'styleguide__palette__heading';
+    wrappingEl.className = 'styleguide__palette__section';
+    headingEl.innerHTML = `${type[0].attribution} colours`;
+    wrappingEl.appendChild(headingEl);
+
+    type.forEach((pair) => {
+      const paletteBlock = document.createElement('div');
+      paletteBlock.className = 'styleguide__palette__block';
```
Comment (jamesgorrie) [inline client/js/components/palette.js]: We should think about stylename injection (modules) to reduce replication.

File: client/js/components/palette.js (hunk 2 above)
Comment (jamesgorrie) [inline client/js/components/palette.js]: And could be used elsewhere, @gestchild where was it again we were looking at doing that?

File: client/scss/styleguide/_styleguide.scss (hunk 1 above)
Comment (davidpmccormick) [inline client/scss/styleguide/_styleguide.scss]: The key benefit of BEM as I see it is to minimise reliance on the cascade, in order to minimise issues arising from classes with more specificity than necessary, so avoiding nesting where possible is a good thing.

You're right, it's slightly moot here, but I think there's still a benefit in having any styleguide-specific classes prefixed, so there's absolutely no ambiguity in our markup regarding what is and isn't required in the app and styleguide.

File: client/js/components/palette.js (hunk 2 above)
Comment (davidpmccormick) [inline client/js/components/palette.js]: Aye – think it might have been number of columns at specific breakpoints?

**Comment by davidpmccormick** [comment]
I haven't got a clear answer on what 'primary' and 'secondary' refer to from the C&T doc, but I'd hazard a guess at it being a distinction between colours for text/background for long-form copy (primary), and everything else (secondary).

**Comment by jamesgorrie** [comment]
AH, okay - it'd be great to get clarity (,to bake the logic into our code) on that and reasoning too (testing). @jennpb could we get this?

File: client/js/components/palette.js (hunk 2 above)
Comment (gestchild) [inline client/js/components/palette.js]: Yes, columns stuff and typography values

File: client/scss/styleguide/_styleguide.scss (hunk 1 above)
Comment (jamesgorrie) [inline client/scss/styleguide/_styleguide.scss]: It still feels like metadata about the object that should not be incapsulated into it's style / classname. This is a coupling, which in itself sets off warning signs. I image if we had two `.palette__name`s, one in the site, and one in the styleguide, you'd probably set the base, and set the differences in a `palette__name--styleguide` modifier.

In short I wonder about storing the metadata that this element only exists in the styleguide in the classname (and superfluous metadata as it's not used for anything).

We could create a separate `styleguide.scss` file that is only included on the pattern pages which is a more explanitory I would imagine (and get's rid of the ambiguity), but also overkill in this case?

But true, ambiguity is bad.

**Comment by jamesgorrie** [comment]
👍 still...

**Comment by jennpb** [comment]
I'll ask C&T to clarify their use cases for Primary and Secondary.

**Comment by jennpb** [comment]
From Gus: 

> Hey @gus @samgeorge . Can we confirm the use cases for Primary and Secondary colours in the Palette? Is it fair to say that the Primary colours are the basis for the simplest manifestation of text and actions? And the Secondary colours are there to make up the combined elements of the site: the breakers, cards, transporters, etc...?

gus [11:17 AM]  
hi jenn, more or less

new messages
[11:18]  
I wouldn’t get too hung up on the difference between primary and secondary. It’s more a case that the primary colours appear way more often than the secondary.

[11:20]  
primary could be viewed as the base, the core elements. secondary perhaps as more bespoke, to be used here and there to accent stuff (and for particular reasons)

**Comment by jamesgorrie** [comment]
@davidpmccormick it might be nice to try and find a way to encapsulate what @jennpb has got from C&T in code, not sure how though...

**Pull request merged**